diff --git a/node_modules/jsdom/lib/jsdom/living/helpers/create-element.js b/node_modules/jsdom/lib/jsdom/living/helpers/create-element.js
index eef3a9a..0840168 100644
--- a/node_modules/jsdom/lib/jsdom/living/helpers/create-element.js
+++ b/node_modules/jsdom/lib/jsdom/living/helpers/create-element.js
@@ -23,7 +23,7 @@ const INTERFACE_TAG_MAPPING = {
     HTMLElement: [
       "abbr", "address", "article", "aside", "b", "bdi", "bdo", "cite", "code", "dd", "dfn", "dt", "em", "figcaption",
       "figure", "footer", "header", "hgroup", "i", "kbd", "main", "mark", "nav", "noscript", "rp", "rt", "ruby", "s",
-      "samp", "section", "small", "strong", "sub", "summary", "sup", "u", "var", "wbr"
+      "samp", "search", "section", "small", "strong", "sub", "summary", "sup", "u", "var", "wbr"
     ],
     HTMLAnchorElement: ["a"],
     HTMLAreaElement: ["area"],
